CMakeLists.txt

```diff
@@ -444,6 +444,12 @@ if(__BUILD_FOR_R)
   endif(MSVC)
 endif(__BUILD_FOR_R)
 
+# fmtlib/fmt
+add_subdirectory(external_libs/fmt)
+set_property(TARGET fmt PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
Hmm, I can't find info about `POSITION_INDEPENDENT_CODE` for CMake 2.8: https://cmake.org/cmake/help/v2.8.0/cmake.html. I guess it was introduced later.
But it exists in `3.0` version: https://cmake.org/cmake/help/v3.0/prop_tgt/POSITION_INDEPENDENT_CODE.html.

My personal opinion is that CMake 2.8 is very old and we can bump minimum version to `3.0`.
https://github.com/microsoft/LightGBM/blob/2903ca0c9fc4b5cb55dbb6ef79c3a53f8fd413a2/CMakeLists.txt#L7-L9

Or we should find another portable solution for `fmt` (roll back to previous one with `CMAKE_FLAGS`).

@guolinke @chivee @btrotta

Reply:
updating CMake version is okay for me.